libs/ui-components/src/hooks/useAlerts.ts

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import { useAppContext } from './useAppContext';
+import { useAccessReview } from './useAccessReview';
+import { RESOURCE, VERB } from '../types/rbac';
+
+// AlertManager alert structure
+type AlertManagerAlert = {
+  fingerprint: string;
+  labels: {
+    alertname: string;
+    org_id: string;
+    resource: string;
+    [key: string]: string;
+  };
+  annotations: Record<string, string>;
+  startsAt: string;
+  endsAt: string;
+  updatedAt: string;
+  status: {
+    state: string;
+    inhibitedBy: string[];
+    mutedBy: string[];
+    silencedBy: string[];
+  };
+  receivers: Array<{ name: string }>;
+};
+
+const DEFAULT_REFRESH_INTERVAL = 30000; // 30 seconds
+
+export const useAlerts = (
```

Comment:
why are we not reusing `useFetchPeriodically` hook ?

Reply:
I did it initially because I was using this hook to detect when alerts were disabled (we know in the UI as it returns 501), and in that case we didn't want to keep fetch .

Since later I introduced "useAlertsEnabled", the main hook can use useFetchPeriodically. Will refactor later.